frontend/src/pages/jio/index.tsx

```diff
@@ -0,0 +1,89 @@
+import { useEffect } from 'react';
+import { Typography } from '@mui/material';
+import { Container, Stack } from '@mui/system';
+import { useRequest } from 'ahooks';
+import { useNavigate, useParams } from 'react-router-dom';
+import FloatingBottomCard from '../../components/FloatingBottomCard';
+import LoadingScreen from '../../components/LoadingScreen';
+import useCustomSnackbar from '../../hooks/useCustomSnackbar';
+import Logo from '../../components/Logo';
+import Page from '../../components/Page';
+import { getJio } from '../../services/jios';
+import LoginForm from '../auth/LoginForm';
+import JioCard from '../dashboard/jios/list/allJios/JioCard';
+import Page404 from '../error/Page404';
+import { useDispatch } from '../../store';
+import { setRedirectPath } from '../../store/reducers/redirectPath';
+import { PATH_USER } from '../../routes/paths';
+import { UserProfileLocationState } from '../publicProfile';
+import useAuthState from '../../hooks/auth/useAuthState';
+
+export default function JioPage() {
+  const { authenticated } = useAuthState();
+  const { id } = useParams();
+  const snackbar = useCustomSnackbar();
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const jioId = id as string;
+
+  const { data, loading } = useRequest(() => getJio(jioId), {
+    onError: () => {
+      snackbar.enqueueError('Failed to retrieve Jio.');
+    },
+  });
+
+  useEffect(() => {
+    if (data && !data.data.isClosed) {
+      const redirectPathTo = PATH_USER.general.user(data.data.creatorId);
+      const redirectPathOptions = {
+        state: {
+          user: data.data.creatorProfile,
+          isShowFloatingButton: true,
+        } as UserProfileLocationState,
+      };
+
+      if (authenticated) {
```

Comment:
alternative (cleaner way to do this):
1. In useAuthenticated, create a new parameter called redirectTo
![image](https://user-images.githubusercontent.com/24221801/198869128-de2ae625-0fee-4fe3-8b50-732f3bbb3734.png)
2. pass the redirectTo prop to like so:
`checkAuth({redirectTo})`
3. at the start of the component (line 22), just call `useAuthenticated(redirectToPath)`